pallets/fiat_filter/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+
+use common::PriceProvider;
+use core_mods::{anchor, attest, blob, did, revoke};
+use frame_support::{
+    decl_error, decl_module,
+    dispatch::{
+        DispatchError, DispatchErrorWithPostInfo, DispatchResultWithPostInfo, PostDispatchInfo,
+        UnfilteredDispatchable,
+    },
+    traits::{Currency, ExistenceRequirement, IsSubType, WithdrawReasons},
+    weights::{GetDispatchInfo, Pays},
+    Parameter,
+};
+use frame_system::{self as system, ensure_signed};
+use sp_std::boxed::Box;
+
+#[cfg(test)]
+#[allow(non_snake_case)]
+mod tests;
+
+/// The pallet's configuration trait
+/// Configure the pallet by specifying the parameters and types on which it depends.
+pub trait Config:
+    system::Config + did::Trait + anchor::Trait + blob::Trait + revoke::Trait + attest::Trait
+{
+    /// Config option for updating the DockFiatRate
+    type PriceProvider: common::PriceProvider;
+    /// The module's Currency type definition
+    type Currency: Currency<Self::AccountId>;
+
+    /// The outer call (dispatchable) that this module is called with. It is possible to use another type here, but
+    /// it's expected that your runtime::Call will be used.
+    /// Since we need to match by origin crate, we implement `IsSubType` for each Call type we want to match for.
+    type Call: Parameter
+        + UnfilteredDispatchable<Origin = Self::Origin>
+        + GetDispatchInfo
+        + IsSubType<did::Call<Self>>
+        + IsSubType<anchor::Call<Self>>
+        + IsSubType<blob::Call<Self>>
+        + IsSubType<revoke::Call<Self>>
+        + IsSubType<attest::Call<Self>>;
+}
+
+decl_error! {
+    pub enum Error for Module<T: Config> where BalanceOf<T>: From<u64> {
+        /// Failed calculation because of numeric type overflow
+        ArithmeticOverflow,
+        /// checked_div: tried to divide by zero
+        DivideByZero,
+        /// Call is not among the core_mods ones that should go through fiat_filter
+        UnexpectedCall,
+        /// Call to T::PriceProvider::optimized_get_dock_usd_rate() failed
+        NoPriceFound,
+    }
+}
+
+// The module's callable functions (Dispatchable functions)
+// Dispatchable functions must be annotated with a weight and must return a DispatchResult.
+decl_module! {
+    pub struct Module<T: Config> for enum Call
+    where
+        origin: T::Origin,
+        BalanceOf<T>: From<u64>,
+    {
+        // Errors must be initialized if they are used by the pallet.
+        type Error = Error<T>;
+
+        /// Execute a Call. Must be a core_mods call (DID, Blob, Anchor, Revoke/Unrevoke, Attest)
+        #[weight = 10_000]
+        pub fn execute_call(origin, call: Box<<T as Config>::Call>) -> DispatchResultWithPostInfo {
+            Ok(Self::execute_call_(origin, &call)?)
+        }
+    }
+}
+
+type BalanceOf<T> =
+    <<T as Config>::Currency as Currency<<T as system::Config>::AccountId>>::Balance;
+
+type AmountUsd = u64;
+
+// all prices given in nUSD (billionth USD)
```

Comment:
I bumped up the unit to avoid multiplication later in `compute_call_fee_dock_()`
**Is this fine ?**
For sure I can add a comment justifying that (if we keep it this way)

Reply:
Got it. Comment here would be great.